### PR TITLE
EVM Staking - Nomination Flow

### PR DESCRIPTION
**Pull request by devpavan04**
## Summary of changes

- EVM users can now bond tokens and nominate validators.

### Proposed area of change

- [ ] `apps/bridge-dapp`
- [ ] `apps/hubble-stats`
- [ ] `apps/stats-dapp`
- [x] `apps/tangle-dapp`
- [ ] `apps/testnet-leaderboard`
- [ ] `apps/faucet`
- [x] `libs/webb-ui-components`

### Screen Recording (UI Implementation)

https://github.com/webb-tools/webb-dapp/assets/53374218/e3bae463-30a2-4e79-ad48-f384ebe13c1a

### CODE REVIEW

File: apps/tangle-dapp/constants/contract.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
+export const StakinInterfaceAddress =
```
Comment (drewstone) [inline apps/tangle-dapp/constants/contract.ts:1]: `StakingInterfacePrecompileAddress`?

File: apps/tangle-dapp/constants/evm.ts [hunk 2]
```diff
@@ -8,76 +8,118 @@ import {
...
-      console.log('nominateTx', nominateTx);
+    const txHash = await evmWalletClient.writeContract(request);
 
-      return nominateTx;
-    }
-  } catch (e) {
-    console.log(e);
+    return txHash;
+  } catch (e: any) {
+    throw new Error(e);
```
Comment (drewstone) [inline apps/tangle-dapp/constants/evm.ts:123]: This doesn't seem like proper error handling

Comment (drewstone) [reply apps/tangle-dapp/constants/evm.ts:123]: If you're just going to throw an error it's no different than removing the `try/catch` altogether

File: apps/tangle-dapp/constants/contract.ts (hunk 1 above)
Comment (devpavan04) [reply apps/tangle-dapp/constants/contract.ts:1]: I'll update

File: apps/tangle-dapp/constants/polkadot.ts [hunk 3]
```diff
@@ -142,3 +142,21 @@ export const getValidatorIdentity = async (
     );
   }
 };
+
+export const isNominatorAlreadyBonded = async (
```
Comment (drewstone) [inline apps/tangle-dapp/constants/polkadot.ts:146]: Does this really need a try/catch? It seems like you wrap it in a try/catch in the component the calls it.

**Review by drewstone — CHANGES_REQUESTED** [review]
Would be curious about improving the error handling. Seems like try/catches are becoming used repeatedly. It might be overused in some places.

File: apps/tangle-dapp/constants/evm.ts (hunk 2 above)
Comment (devpavan04) [reply apps/tangle-dapp/constants/evm.ts:123]: @drewstone We catch this error to notify users in this function [here](https://github.com/webb-tools/webb-dapp/blob/71f8eab8a53b4be8d9b0964834e5d09bccf853da/apps/tangle-dapp/containers/DelegateTxContainer/DelegateTxContainer.tsx#L105-L165). But I'll optimize the use of try/catch.

**Comment by devpavan04** [comment]
Closing this since the EVM Staking app will be migrated to a new React app.

**Pull request closed without merging**